ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -448,9 +449,10 @@ impl Module {
         self.params.push(loc);
     }
 
-    /// Return a slice over all trace parameters.
-    pub(crate) fn params(&self) -> &[yksmp::Location] {
-        &self.params
+    /// Return the parameter at a given [InstIdx].
+    pub(crate) fn param(&self, iidx: InstIdx) -> &yksmp::Location {
```

Comment:
I'm not sure about making this `InstIdx`. My understanding is that an `InstIdx` is only used to index into the `insts` vector of the module. Should this not be a new type `ParamIdx`, as it's indexing into the `params` vector?

Reply:
Interestingly, I originally made it `ParamIdx`, got almost as far as a PR then.... I realised that the if we have *n* parameters, the first *n* instructions must be parameters. It's baked into our system that `%0`...`%n` are `ParameterInst`s, so introducing `ParamIdx` meant I just ended up with (infallible) conversions to/from `InstIdx` without really doing much useful.

Reply:
Ok.